android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -627,7 +675,19 @@ private void isEnabled(Result poResult) {
     /// Method to connect/disconnect wifi service
     private void setEnabled(MethodCall poCall, Result poResult) {
         Boolean enabled = poCall.argument("state");
-        moWiFi.setWifiEnabled(enabled);
+
+        /**
+         * Open native Android WiFi settings on android Q and later
+         */
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q) {
+            moWiFi.setWifiEnabled(enabled);
+        }
+            else {
+                Intent wifiStateIntent = new Intent(Settings.Panel.ACTION_WIFI);
+                wifiStateIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                this.moContext.startActivity(wifiStateIntent);
+            }
+
```

Comment:
This could break for some implementation of `setEnabled`, for those not expecting "user setting to open".

For eg, in my app - I just "disable" and "unable" wifi once after calling `disconnect` to make sure that Android "reconnects back to main network".

Can we come up with some bool arg that handles this case? without breaking current implmentation.

Reply:
As I understand, the `setWifiEnabled` was deprecated in API level 29 or android Q.

In that case, we can add an optional argument in `setEnabled` to handle whether to open native android WiFi settings or not in API level 29 and above.

Reply:
yes, that is what I meant, add something like `bool shouldOpenSettings: false` param to `setEnabled`.

Reply:
@daadu Could you please check the newly added feature? Thanks